cornice/service.py

```diff
@@ -465,9 +465,11 @@ def wrapper(request):
         ob = None
         view_ = view
         if 'klass' in args and not callable(view):
+            #XXX: given that request.context exists and root-factory only expects request param, having params seems unnecessary
```

Comment:
If this was done, `__init__()` would not need to be overloaded by decorated class. For the time being, it's left as is so decorated classes must declare `context` as optional param.

In current test-cases, `ThingImp` and `UserImp` have `context` as optional param in their `__init__` methods.

See:

https://github.com/Cornices/cornice/blob/76b0390a333d59b0977646aacb76809c7829d3ad/tests/test_imperative_resource.py#L30

https://github.com/Cornices/cornice/blob/76b0390a333d59b0977646aacb76809c7829d3ad/tests/test_imperative_resource.py#L40